src/main/java/fi/metatavu/metamind/bot/functions/BeerApiBeerInfoMetaBotFunction.java

```diff
@@ -0,0 +1,164 @@
+package fi.metatavu.metamind.bot.functions;
+
+import java.text.NumberFormat;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.bladecoder.ink.runtime.Story;
+import com.rabidgremlin.mutters.bot.ink.CurrentResponse;
+import com.rabidgremlin.mutters.core.IntentMatch;
+import com.rabidgremlin.mutters.core.session.Session;
+
+import org.apache.commons.lang3.math.NumberUtils;
+import org.slf4j.Logger;
+
+import fi.metatavu.beer.ApiClient;
+import fi.metatavu.beer.client.BarBeersApi;
+import fi.metatavu.beer.client.Beer;
+import fi.metatavu.beer.client.BeerImage;
+import fi.metatavu.beer.client.BeerRating;
+import fi.metatavu.metamind.story.StoryController;
+
+/**
+ * Metabot function to get single beer info from Beer API
+ * 
+ * @author Heikki Kurhinen
+ */
+@ApplicationScoped
+public class BeerApiBeerInfoMetaBotFunction extends AbstractMetaBotFunction {
+
+  @Inject
+  private Logger logger;
+
+  @Inject
+  private StoryController storyController;
+
+  @Override
+  public String getFunctionName() {
+    return "BEER_API_BEER_INFO";
+  }
+
+  @Override
+  public void execute(CurrentResponse currentResponse, Session session, IntentMatch intentMatch, Story story, String param) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)